pkg/gpu/probe_test.go

```diff
@@ -12,49 +12,66 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 
+	"github.com/DataDog/datadog-agent/pkg/ebpf/ebpftest"
 	"github.com/DataDog/datadog-agent/pkg/gpu/config"
 	"github.com/DataDog/datadog-agent/pkg/gpu/testutil"
 	"github.com/DataDog/datadog-agent/pkg/network/usm/utils"
 	"github.com/DataDog/datadog-agent/pkg/process/monitor"
 )
 
-func TestProbeCanLoad(t *testing.T) {
+type probeTestSuite struct {
+	suite.Suite
+}
+
+func TestProbe(t *testing.T) {
 	if err := config.CheckGPUSupported(); err != nil {
 		t.Skipf("minimum kernel version not met, %v", err)
 	}
 
+	ebpftest.TestBuildModes(t, []ebpftest.BuildMode{ebpftest.CORE}, "", func(t *testing.T) {
+		suite.Run(t, new(probeTestSuite))
+	})
+}
+
+func (s *probeTestSuite) getProbe() *Probe {
+	t := s.T()
+
 	cfg := config.New()
+
+	// Avoid waiting for the initial sync to finish in tests, we don't need it
```

Comment:
do we want to have at least one test to check this scenario and make sure there are no "timing" edge-cases or any other unexpected surprises?

Reply:
We could check it, although it shouldn't be too problematic, the initial sync is just scanning the procfs before starting the goroutine that listens to new processes. [Created a ticket to track it](https://datadoghq.atlassian.net/browse/EBPF-614)